modules/secure-cloud-run-core/main.tf

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "cloud_run" {
+  source  = "GoogleCloudPlatform/cloud-run/google"
+  version = "~> 0.3.0"
+
+  service_name          = var.service_name
+  project_id            = var.serverless_project_id
+  location              = var.location
+  image                 = var.image
+  service_account_email = var.cloud_run_sa
+  encryption_key        = var.encryption_key
+  members               = var.members
+  env_vars              = var.env_vars
+
+  service_annotations = {
+    "run.googleapis.com/ingress" = "internal-and-cloud-load-balancing"
+  }
+
+  template_annotations = {
+    "autoscaling.knative.dev/maxScale" : 2,
```

Comment:
Would a user want to control scaling?

Reply:
@bharathkkb maybe we could have a input with a map of user defined template_annotations and merge it with a local one that has 

```hcl
locals {
  required_template_annotations = {
    "run.googleapis.com/vpc-access-connector" = var.vpc_connector_id,
    "run.googleapis.com/vpc-access-egress"    = "private-ranges-only"
  }
}

Reply:
I like the idea but I am wary of users have to remember magic strings like `autoscaling.knative.dev/maxScale` for controlling something as simple as scaling. I suggest for the most common changes like scaling we surface top level vars and in the future we can do map merge if we need to support additional changes.

Reply:
I've opened an issue to address this discussion: https://github.com/GoogleCloudPlatform/terraform-google-cloud-run/issues/46